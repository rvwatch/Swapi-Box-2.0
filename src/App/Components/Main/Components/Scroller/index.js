import React from 'react';
import { string } from 'prop-types';
import './Scroller.css';

const Scroller = ({ title, opening_crawl }) => {
  return (
    <section className="star-wars">
      <div className="fade" />
      <div className="crawl">
        <div className="title">
          <h1>{title}</h1>
        </div>
        <p>{opening_crawl}</p>
      </div>
    </section>
  );
};

Scroller.propTypes = {
  title: string,
  opening_crawl: string
};

export default Scroller;
