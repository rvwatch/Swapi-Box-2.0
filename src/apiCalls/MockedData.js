export const mockUrl = 'starwars.com';

export const mockUrlArray = ['starwars.com', 'starwars.com', 'starwars.com'];

export const mockSingleFilm = {
  title: 'Movie!',
  opening_crawl: 'Movie text!'
};

export const mockApiFilm = {
  results: [
    { title: 'Movie!', opening_crawl: 'Movie text!' },
    { title: 'Movie!', opening_crawl: 'Movie text!' },
    { title: 'Movie!', opening_crawl: 'Movie text!' }
  ]
};

export const mockPerson = {
  name: 'Luke Skywalker',
  homeworld: 'Tatooine',
  population: 200000,
  species: 'Human'
};

export const mockApiPeople = {
  results: [
    {
      name: 'Luke Skywalker',
      homeworld: 'Tatooine',
      population: 200000,
      species: 'Human'
    },
    {
      name: 'C-3PO',
      homeworld: 'Tatooine',
      population: 200000,
      species: 'Droid'
    },
    {
      name: 'Obi-Wan Kenobi',
      homeworld: 'Stewjon',
      population: 'unknown',
      species: 'Human'
    }
  ]
};

export const mockCleanPeople = [
  {
    homeworld: undefined,
    id: 'Luke Skywalker0',
    name: 'Luke Skywalker',
    population: undefined,
    species: undefined
  },
  {
    homeworld: undefined,
    id: 'C-3PO1',
    name: 'C-3PO',
    population: undefined,
    species: undefined
  },
  {
    homeworld: undefined,
    id: 'Obi-Wan Kenobi2',
    name: 'Obi-Wan Kenobi',
    population: undefined,
    species: undefined
  }
];

export const mockPlanetData = {
  results: [
    {
      name: 'Tatooine',
      rotation_period: '23',
      orbital_period: '304',
      diameter: '10465',
      climate: 'arid',
      gravity: '1 standard',
      terrain: 'desert',
      surface_water: '1',
      population: '200000',
      residents: ['']
    },
    {
      name: 'Tatooine',
      rotation_period: '23',
      orbital_period: '304',
      diameter: '10465',
      climate: 'arid',
      gravity: '1 standard',
      terrain: 'desert',
      surface_water: '1',
      population: '200000',
      residents: ['']
    },
    {
      name: 'Tatooine',
      rotation_period: '23',
      orbital_period: '304',
      diameter: '10465',
      climate: 'arid',
      gravity: '1 standard',
      terrain: 'desert',
      surface_water: '1',
      population: '200000',
      residents: ['']
    }
  ]
};

export const mockCleanPlanets = [
  {
    climate: 'arid',
    id: 'Tatooine0',
    name: 'Tatooine',
    population: '200000',
    residents: [{ name: undefined }],
    terrain: 'desert'
  },
  {
    climate: 'arid',
    id: 'Tatooine1',
    name: 'Tatooine',
    population: '200000',
    residents: [{ name: undefined }],
    terrain: 'desert'
  },
  {
    climate: 'arid',
    id: 'Tatooine2',
    name: 'Tatooine',
    population: '200000',
    residents: [{ name: undefined }],
    terrain: 'desert'
  }
];

export const mockResidents = [
  { name: 'Luke Skywalker' },
  { name: 'Luke Skywalker' },
  { name: 'Luke Skywalker' }
];

export const mockVehicleData = {
  results: [
    {
      name: 'Sand Crawler',
      model: 'Digger Crawler',
      manufacturer: 'Corellia Mining Corporation',
      cost_in_credits: '150000',
      length: '36.8',
      max_atmosphering_speed: '30',
      crew: '46',
      passengers: '30',
      cargo_capacity: '50000',
      consumables: '2 months',
      vehicle_class: 'wheeled',
      pilots: [],
      films: ['https://swapi.co/api/films/5/', 'https://swapi.co/api/films/1/'],
      created: '2014-12-10T15:36:25.724000Z',
      edited: '2014-12-22T18:21:15.523587Z',
      url: 'https://swapi.co/api/vehicles/4/'
    }
  ]
};

export const mockCleanVehicle = [
  {
    id: 'Sand Crawler0',
    model: 'Digger Crawler',
    name: 'Sand Crawler',
    passengers: '30',
    vehicle_class: 'wheeled'
  }
];

export const mockSelectedCard1 = {
  name: 'Luke Skywalker',
  homeworld: 'Tatooine',
  population: 200000,
  species: 'Human',
  id: 1
};

export const mockSelectedCard2 = {
  name: 'Luke Skywalker',
  homeworld: 'Tatooine',
  population: 200000,
  species: 'Human',
  id: 2
};

export const mockSelectedCard3 = {
  name: 'Luke Skywalker',
  homeworld: 'Tatooine',
  population: 200000,
  species: 'Human',
  id: 3
};

export const mockFavorites = [
  {
    name: 'Luke Skywalker',
    homeworld: 'Tatooine',
    population: 200000,
    species: 'Human',
    id: 1
  },
  {
    name: 'Luke Skywalker',
    homeworld: 'Tatooine',
    population: 200000,
    species: 'Human',
    id: 2
  },
  {
    name: 'Luke Skywalker',
    homeworld: 'Tatooine',
    population: 200000,
    species: 'Human',
    id: 3
  }
];

export const mockStarshipData = {
  results: [
    {
      MGLT: '40',
      cargo_capacity: '250000000',
      consumables: '6 years',
      cost_in_credits: '1143350000',
      created: '2014-12-15T12:31:42.547000Z',
      crew: '279144',
      edited: '2017-04-19T10:56:06.685592Z',
      films: (2)[
        ('https://swapi.co/api/films/2/', 'https://swapi.co/api/films/3/')
      ],
      hyperdrive_rating: '2.0',
      length: '19000',
      manufacturer: 'Kuat Drive Yards, Fondor Shipyards',
      max_atmosphering_speed: 'n/a',
      model: 'Executor-class star dreadnought',
      name: 'Executor',
      passengers: '38000',
      pilots: [],
      starship_class: 'Star dreadnought',
      url: 'https://swapi.co/api/starships/15/'
    }
  ]
};

export const mockCleanStarship = [
  {
    crew: '279144',
    id: 'Executor0',
    length: '19000',
    model: 'Executor-class star dreadnought',
    name: 'Executor',
    passengers: '38000',
    starship_class: 'Star dreadnought'
  }
];
