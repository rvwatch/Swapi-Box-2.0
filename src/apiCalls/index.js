export const getData = async url => {
  try {
    const response = await fetch(url);
    const starWarsData = await response.json();
    return starWarsData;
  } catch (errs) {
    throw new Error('getData has thrown an error');
  }
};

export const getScrollText = async url => {
  const filmData = await getData(url);
  const selectFilm = Math.floor(Math.random() * filmData.results.length);
  const { title, opening_crawl } = filmData.results[selectFilm];
  return { title, opening_crawl };
};

export const getPeopleData = async url => {
  const peopleData = await getData(url);

  const peoplePromises = peopleData.results.map(async (person, index) => {
    const homeworld = await getData(person.homeworld);
    const species = await getData(person.species);
    return {
      name: person.name,
      homeworld: homeworld.name,
      population: homeworld.population,
      species: species.name,
      id: person.name + index
    };
  });
  return Promise.all(peoplePromises);
};

export const getPlanetData = async url => {
  const planetData = await getData(url);

  const planetPromises = planetData.results.map(async (planet, index) => {
    const { name, terrain, population, climate, residents } = planet;
    const residentList = await getResidents(residents);
    return {
      name,
      terrain,
      population,
      climate,
      residents: [...residentList],
      id: name + index
    };
  });
  return Promise.all(planetPromises);
};

export const getResidents = async residentUrls => {
  const residentPromises = residentUrls.map(async url => {
    const person = await getData(url);
    return { name: person.name };
  });
  return Promise.all(residentPromises);
};

export const getVehicleData = async url => {
  const vehicleData = await getData(url);

  const vehiclePromises = vehicleData.results.map(async (vehicle, index) => {
    const { name, model, vehicle_class, passengers } = vehicle;
    return {
      name,
      model,
      vehicle_class,
      passengers,
      id: name + index
    };
  });
  return Promise.all(vehiclePromises);
};

export const getStarshipData = async url => {
  const starshipData = await getData(url);

  const starshipPromises = starshipData.results.map(async (ship, index) => {
    const { name, model, starship_class, passengers, crew, length } = ship;
    return {
      name,
      model,
      starship_class,
      passengers,
      crew,
      length,
      id: name + index
    };
  });
  return Promise.all(starshipPromises);
};
