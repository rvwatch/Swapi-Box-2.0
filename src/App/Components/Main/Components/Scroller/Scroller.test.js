import React from 'react';
import { shallow } from 'enzyme';
import Scroller from './index';
import { mockSingleFilm } from '../../../../../apiCalls/MockedData';

describe('Scroller ', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Scroller scrollText={mockSingleFilm} />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
