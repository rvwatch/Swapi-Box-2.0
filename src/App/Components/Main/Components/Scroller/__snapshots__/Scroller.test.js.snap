// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroller  renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scroller
    scrollText={
        Object {
            "opening_crawl": "Movie text!",
            "title": "Movie!",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
/>,
        <div
          className="crawl"
>
          <div
                    className="title"
          >
                    <h1 />
          </div>
          <p />
</div>,
      ],
      "className": "star-wars",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
>
              <h1 />
</div>,
            <p />,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1 />,
              "className": "title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <h1 />
            </div>
            <p />
</div>,
        ],
        "className": "star-wars",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <h1 />
</div>,
              <p />,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1 />,
                "className": "title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
